src/cdk/datetime/datepicker.spec.ts

```diff
@@ -0,0 +1,646 @@
+
+    return TestBed.createComponent(component);
+  }
+
+  describe('with NativeDateModule', () => {
+    describe('standard datepicker', () => {
+      let fixture: ComponentFixture<StandardDatepicker>;
+      let testComponent: StandardDatepicker;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(StandardDatepicker, [NativeDateModule]);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+      }));
+
+      afterEach(fakeAsync(() => {
+        fixture.detectChanges();
+        flush();
+      }));
+
+      it('should initialize with correct value shown in input', () => {
+        expect(fixture.nativeElement.querySelector('input').value).toBe('1/1/2020');
+      });
+
+      it('startAt should fallback to input value', () => {
+        expect(testComponent.datepicker.startAt).toEqual(new Date(2020, JAN, 1));
+      });
+
+      it('should not throw when given wrong data type', () => {
+        testComponent.date = '1/1/2017' as any;
+
+        expect(() => fixture.detectChanges()).not.toThrow();
+      });
+    });
+
+    describe('datepicker with too many inputs', () => {
+      it('should throw when multiple inputs registered', fakeAsync(() => {
+        let fixture = createComponent(MultiInputDatepicker, [NativeDateModule]);
+        expect(() => fixture.detectChanges()).toThrow();
+      }));
+    });
+
+    describe('datepicker with no inputs', () => {
+      let fixture: ComponentFixture<NoInputDatepicker>;
+      let testComponent: NoInputDatepicker;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(NoInputDatepicker, [NativeDateModule]);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+      }));
+
+      afterEach(fakeAsync(() => {
+        fixture.detectChanges();
+      }));
+
+      it('should not throw when accessing disabled property', () => {
+        expect(() => testComponent.datepicker.disabled).not.toThrow();
+      });
+    });
+
+    describe('datepicker with startAt', () => {
+      let fixture: ComponentFixture<DatepickerWithStartAt>;
+      let testComponent: DatepickerWithStartAt;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(DatepickerWithStartAt, [NativeDateModule]);
+        fixture.detectChanges();
+
+        testComponent = fixture.componentInstance;
+      }));
+
+      afterEach(fakeAsync(() => {
+        fixture.detectChanges();
+      }));
+
+      it('explicit startAt should override input value', () => {
+        expect(testComponent.datepicker.startAt).toEqual(new Date(2010, JAN, 1));
+      });
+    });
+
+    describe('datepicker with ngModel', () => {
+      let fixture: ComponentFixture<DatepickerWithNgModel>;
+      let testComponent: DatepickerWithNgModel;
+
+      beforeEach(fakeAsync(() => {
+        fixture = createComponent(DatepickerWithNgModel, [NativeDateModule]);
+        fixture.detectChanges();
+
+        fixture.whenStable().then(() => {
```

Comment:
In general we try to avoid `whenStable` since it can be flaky when we run the tests across multiple browsers. You should be able to do the same with `tick` or `flush`.